src/libraries/System.Transactions.Local/src/System/Transactions/TransactionManager.cs

```diff
@@ -334,6 +359,29 @@ public static TimeSpan MaximumTimeout
 
                 return s_maximumTimeout;
             }
+            set
+            {
+                TransactionsEtwProvider etwLog = TransactionsEtwProvider.Log;
+                if (etwLog.IsEnabled())
+                {
+                    etwLog.MethodEnter(TraceSourceType.TraceSourceBase, "TransactionManager.set_DefaultMaximumTimeout");
+                }
+
+                if(value < TimeSpan.Zero)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value));
+                }
+
+                s_cachedMaxTimeout = true;
+                s_maximumTimeout = value;
+                s_defaultTimeout = ValidateTimeout(s_defaultTimeout);
+                s_defaultTimeoutValidated = true;
```

Comment:
We should log the same event if the default timeout gets reduced. Save a local copy of s_defaultTimeout before the call to ValidateTimeout, then do a comparison and emit an event similar to the DefaultTimeout setter.

Reply:
Thanks Matt! Updated.